shm/Player.hpp

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include <memory>
+#include "Cargo.hpp"
+#include "Ship.hpp"
+
+class Player
+{
+public:
+    Player(Ship ship, size_t money, size_t space) : money_(money), availableSpace_(space)
```

Comment:
Why do you receive the whole Ship object by copy instead of receiving a pointer?